### PR TITLE
Document volume.kubernetes.io/gid annotation

### PR DESCRIPTION
**Pull request by swagiaal**
Documentation for this feature: https://github.com/kubernetes/kubernetes/pull/20490

### CODE REVIEW

**Comment by chao007** [comment]
Hi,  I think the yaml file "volume.kubernetes.io/gid" should be "pv.beta.kubernetes.io/gid"

File: docs/user-guide/persistent-volumes/walkthrough.md [hunk 1]
```diff
@@ -36,6 +36,31 @@ NAME      LABELS       CAPACITY      ACCESSMODES   STATUS      CLAIM     REASON
...
+```yaml
+apiVersion: v1
+metadata:
+  name: pv1
+  annotations:
+    volume.kubernetes.io/gid: "1234"
+#...
+```
+
+For more information please see [Security Contexts](../security-context.md) for more information.
```
Comment (johndmulhausen) [inline docs/user-guide/persistent-volumes/walkthrough.md]: URL: /docs/user-guide/security-context/

Absolute paths w/no .md suffix, please

File: docs/user-guide/persistent-volumes/walkthrough.md [hunk 2]
```diff
@@ -36,6 +36,31 @@ NAME      LABELS       CAPACITY      ACCESSMODES   STATUS      CLAIM     REASON
 pv0001    type=local   10737418240   RWO           Available 
 ```
 
+### Access Control
+
+Storage configured with GID will only allow writing by pods using the
+same GID. Mismatched or missing GIDs will cause `permission denied`
+errors. Adding GID to a PersistentVolume- via the
```
Comment (johndmulhausen) [inline docs/user-guide/persistent-volumes/walkthrough.md]: PersistenVolume should be surrounded by `backticks`

File: docs/user-guide/persistent-volumes/walkthrough.md [hunk 3]
```diff
@@ -36,6 +36,31 @@ NAME      LABELS       CAPACITY      ACCESSMODES   STATUS      CLAIM     REASON
...
+`pv.beta.kubernetes.io/gid annotation` -allows Kubelet to automatically
```
Comment (johndmulhausen) [inline docs/user-guide/persistent-volumes/walkthrough.md]: Get rid of the dashes before allows and after PersistentVolume

Comment (johndmulhausen) [inline docs/user-guide/persistent-volumes/walkthrough.md]: `kubelet`

File: docs/user-guide/persistent-volumes/walkthrough.md [hunk 4]
```diff
@@ -36,6 +36,31 @@ NAME      LABELS       CAPACITY      ACCESSMODES   STATUS      CLAIM     REASON
...
+add the GID to the pod that requires it. No coordination between an
+admin and end user is required. Setting GID automatically is safe
+because this is only done for PersistentVolumes which are accessed
+through a PersistentVolumeClaim.
```
Comment (johndmulhausen) [inline docs/user-guide/persistent-volumes/walkthrough.md]: `PersistentVolumes`
`PersistenVolumeClaim`

**Comment by saad-ali** [comment]
@pmorie @wongma7 Is this still accurate after https://github.com/kubernetes/kubernetes/pull/29119 and could you please assign someone to take it over.

**Comment by wongma7** [comment]
@pmorie I created a new pr https://github.com/kubernetes/kubernetes.github.io/pull/938

**Comment by pwittrock** [comment]
Closing this since it was moved to  #938

**Pull request closed without merging**